### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by AntonBazhal**

### CODE REVIEW

File: lib/handler.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const _ = require('lodash');
+const aeclient = require('aws-elasticsearch-client');
+const errors = require('./errors');
```
Comment (dougmoscrop) [inline lib/handler.js]: small nitpick, to put local requires together below npm/core requires

File: lib/handler.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  return lambdaHandler((event, context) => {
+    return Promise.resolve().then(() => {
+      if (options.beforeHook) {
+        options.beforeHook(event, context);
+      }
+
+      utils.validate(event, schemas.EVENT, { allowUnknown: true });
+
+      const actions = [];
+      event.Records.forEach(record => {
```
Comment (dougmoscrop) [inline lib/handler.js]: this is effectively `array.reduce` no?

Comment (AntonBazhal) [reply lib/handler.js]: Yeah, good point

File: lib/handler.js (hunk 1 above)
Comment (AntonBazhal) [reply lib/handler.js]: I always did like that at CT, but here I've never seen anyone to follow this practice.. But yes, I prefer to separate requires into to groups.

File: lib/handler.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      if (options.afterHook) {
+        options.afterHook(event, context, result);
+      }
+      return result;
+    })
+    .catch(err => {
+      if (options.errorHook) {
+        return options.errorHook(event, context, err);
+      }
+      throw err;
```
Comment (dougmoscrop) [inline lib/handler.js:100]: It's been a while but is there a difference between rethrowing and instead say, `return Promise.reject(err)` to preserve stack trace?

Comment (AntonBazhal) [reply lib/handler.js:100]: Interesting.. I always thought there is no difference. Will double check.

Comment (AntonBazhal) [reply lib/handler.js:100]: [From some reading](http://stackoverflow.com/questions/33445415/javascript-promises-reject-vs-throw) and my experiments: there is absolutely no difference between two options. Stack is exactly the same.

**Pull request merged**